dkist/net/globus/transfer.py

```diff
@@ -186,37 +201,124 @@ def watch_transfer_progress(task_id, tfr_client, poll_interval=5,
+                if event["is_error"] or verbose:
+                    progress.write(f"{event.get('code', '')}: {event.get('description', '')}")
+                    details = event.get("details", {})
+                    if "error" in details and "body" in details["error"]:
+                        extra_message = details["error"]["body"]
+                        if "context" in details:
+                            context = details["context"][0]
+                            operation = context.get("operation", "")
+                            path = context.get("path", "")
+                            extra_message = f"{operation} | {path}".strip() + " | " + extra_message
+                        progress.write(extra_message)
+
+            # Extract and calculate the transfer speed from the event list
+            speed = (list(map(_get_speed, json_events)) or [None])[0]
+            speed = f"{speed} Mb/s" if speed else ""
+            if speed:
+                progress.set_postfix_str(speed)
+
+            # Get the status of the task to see how many files we have processed.
+            task = tfr_client.get_task(task_id)
+            status = task['status']
+            progress.total = task['files']
+            progress.update((task['files_skipped'] + task['files_transferred']) - progress.n)
+
+            # If the status of the task is not active we are finished.
+            if status != "ACTIVE":
+                progress.write(f"Task completed with {status} status.")
+                progress.close()
+                break
+
+            # Wait for next poll
+            time.sleep(poll_interval)
+
+    except KeyboardInterrupt:
+        progress.write("Cancelling Task")
+        task = tfr_client.cancel_task(task_id)
+        progress.close()
+
+
+def _orchestrate_transfer_task(file_list: List[PathLike],
+                               recursive: List[bool],
+                               destination_path: PathLike = "/~/",
+                               destination_endpoint: str = None,
+                               *,
+                               progress: Union[bool, Literal["verbose"]] = True,
+                               wait: bool = True):
+    """
+    Transfer the files given in file_list to the path on ``destination_endpoint``.
+
+    Parameters
+    ----------
+    file_list
+
+    recursive
+
+    destination_path
+
+    destination_endpoint
+        A unique specifier for a Globus endpoint. If `None` a local
+        endpoint will be used if it can be found, otherwise an error will
+        be raised. See `~dkist.net.globus.get_endpoint_id` for valid
+        endpoint specifiers.
+
+    progress
+        If `True` status information and a progress bar will be displayed
+        while waiting for the transfer to complete.
+        If ``progress="verbose"`` then all globus events generated during the
+        transfer will be shown (by default only error messages are shown.)
```

Comment:
I think a sentence at the end saying something like "This keyword only matters if ``wait=True`` will help ensure people understand this behavior.